### PR TITLE
Repo page displays fork status

### LINKED ISSUE #47546: Add a visual indicator to the repository page if a repo is a fork
- [ ] Maybe in one of these areas when a repo is visited:

image

- [ ] Maybe on repo settings page

screenshot_2023-02-13_15 52 55@2x

- [ ] Stretch goal: display which repository the fork is a fork of

Needs design input.

/cc @ryphil

### PR DESCRIPTION
**Pull request by cbart**
Closes #47546

We are adding a visual indicator that a repo is a fork.

https://www.loom.com/share/c2ca528d704a41f8865c7dff62d2753c

Not fork:

Screenshot 2023-02-17 at 1 52 57 PM

Fork:

Screenshot 2023-03-15 at 9 26 14 PM

Two elements change when repo is a fork:

- The icon on the left hand side of the repo title H2 becomes `mdiSourceFork` while normally it is `mdiSourceRepo`.
- The repo title gets a secondary outline badge on the right hand side that says _Fork_.

## Test plan

Unit test verifies we display the fork badge.

Believe it or not, this is the first unit test for the treePage component.

## App preview:

- [Web](https://sg-web-cbart-display-fork-status-repo.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.01 kb) | 0.00% (+0.01 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/cbart/display-fork-status-repo-page/compare-report.html;tab=live_object) for a full comparison between the commits 22f2b65 and 832b62c or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/tree/TreePage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            expect(screen.queryByTestId('repo-fork-badge')).not.toBeInTheDocument()
+        })
+
+        it('displays a page that is a fork', () => {
+            const repo = repoDefaults()
+            repo.isFork = true
+            const props = treePagePropsDefaults(repo)
+            renderWithBrandedContext(<TreePage {...props} />)
+            screen.debug()
+            expect(screen.queryByTestId('repo-fork-badge')).toBeInTheDocument()
```
Comment (cbart) [inline client/web/src/repo/tree/TreePage.test.tsx:84]: `queryByTestId` returns null

File: client/web/src/repo/tree/TreePage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            renderWithBrandedContext(<TreePage {...props} />)
+            expect(screen.queryByTestId('repo-fork-badge')).not.toBeInTheDocument()
+        })
+
+        it('displays a page that is a fork', () => {
+            const repo = repoDefaults()
+            repo.isFork = true
+            const props = treePagePropsDefaults(repo)
+            renderWithBrandedContext(<TreePage {...props} />)
+            screen.debug()
```
Comment (cbart) [inline client/web/src/repo/tree/TreePage.test.tsx:83]: this does not print anything anywhere that I can see if I run it in VSCode or in `pnpm test`

**Review by varsanojidan — APPROVED** [review]
Nice! 

Side note: this is unrelated to your change because I know this is what it looks like currently, but I kind of wish the non-fork icon was something else, not really a fan of it. Also the fact that it’s filled-in in comparison to the fork one kinda bugs me 😅

**Review by mrnugget — APPROVED** [review]
Very nice! In your screenshot there's a `/` to the left of the `Fork` icon that I find a bit confusing, but I can't find it in the code?

**Comment by cbart** [comment]
> Very nice! In your screenshot there's a / to the left of the Fork icon that I find a bit confusing, but I can't find it in the code?

@limitedmage can you have a look? The repo header was effectively using combo of `<PageHeader.Header>` and `<PageHeader.Breadcrumb>` but there is no path here to take advantage of breadcrumbs. The `Breadcrumb` element (rightly so) adds a `/` with `:not(:last-child)::after { content: ... }` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@14867a4b05e658a94811537ee8f0511e2e9fe248/-/blob/client/wildcard/src/components/PageHeader/Breadcrumb/Breadcrumb.module.scss?L11-17).

What I've done here now is I effectively simplified and inlined the contents of `<PageHeader.Breadcrumb>` plus I added some `ml-2`'s so that it aligns nicely. Not sure if that's kosher.

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Review by limitedmage — APPROVED** [review]
Looks great! Thanks for fixing the slash.

**Pull request merged**